content/release-engineering/codefresh/best-practices.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "Codefresh \"Best Practices\""
+description: "Our tips for working with Codefresh"
+tags:
+- best-practices
+---
+
+# Managing Secrets
+
+Use `chamber` to manage secrets. Provision a dedicated IAM user with limited scope by using the  [terraform-aws-iam-chamber-user](https://github.com/cloudposse/terraform-aws-iam-chamber-user) terraform module. Review our guide on [managing secrets with CI/CD]({{< relref "secrets-management/cicd.md" >}}).
+
+# Naming Kubernetes Contexts
+
+Use cluster name. e.g. `us-west-2-staing-cloudposse-co`
+
+{{< img src="/assets/best-practices-d9e507c5.png" title="Codefresh Kubernetes Integration" >}}
+
+# Use a `codefresh.yml` build file
+
+Always include the pipeline manifest in the source repo. This allows it to be versioned along side the code it builds
+
+{{< img src="/assets/best-practices-70ab60e9.png" title="Use YAML from Repository" >}}
+
+# Stick non-secrets in `codefresh.yml`
+
+Set all non-secret values in the pipeline config so that they can be versioned.
+
+# Use Conditions to Control Flow
+
+Use conditions to control when a build step should be run. This is useful for deploying to various kubernetes clusters by setting the `KUBE_CONTEXT` based on the branch or tag.
+
+```
+when:
+  condition:
+    all:
+      executeForTag: "'${{SEMVERSION_TAG}}' != ''"
+```
+
+# Use a Build Harness
```

Comment:
Why `a Build Harness` and not `the Build Harness`? (there is only one :) 
Probably should remove `a` (and `the`) and use just `Build Harness`.

Reply:
I guess we can. The thing is, that it doesn't have to be ours. A "build harness" is a concept, just like a "Test harness": https://en.wikipedia.org/wiki/Test_harness